tools/toolchain_profiler.py

```diff
@@ -26,10 +26,13 @@ def __init__(self, name):
   def __enter__(self):
     self.start = time.time()
 
-  def __exit__(self, type, value, traceback):
+  def __exit__(self, exc_type, value, traceback):
     # When a block ends debug log the total duration.
     now = time.time()
-    logger.debug('block "%s" took %.2f seconds', self.name, now - self.start)
+    if exc_type:
+      logger.debug('block "%s" raised an exception after %.2f seconds', self.name, now - self.start)
+    else:
+      logger.debug('block "%s" took %.2f seconds', self.name, now - self.start)
```

Comment:
unrelated?

Reply:
Yes.  This just makes debug logging more clear in case of a crash.   Want me to split it out?

Reply:
Seems ok to leave it in to me. Just wanted to check it was intentional.